snapcraft/linters/library_linter.py

```diff
@@ -73,6 +75,24 @@ def run(self) -> List[LinterIssue]:
                 issues=issues,
             )
 
+            # if the elf file is a library, get its resolved path
+            if elf_file.soname or self._is_library_path(elf_file.path):
+                all_libraries.add(str(elf_file.path.resolve()))
+
+            # get paths to the local libraries used by the elf file
+            used_libraries.update(
+                elf_file.load_dependencies(
```

Comment:
Can we obtain the resolved path from the existing list of dependencies instead of computing it all over again?

Reply:
Yes.  It took awhile for me to refactor and eliminate the second computation.  The code is simpler now too.